File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.js
function gcd(a, b) {
    if (b <= 0)
        return a;
    
    return gcd(b, a % b);
}

function lcm(a, b) {
    return a * b / gcd(a, b);
}

function solution(arr) {
    let result;
    if (arr.length === 1) {
        return arr[0];
    } else if (arr.length === 2) {
        return lcm (arr[0], arr[1]);
    } else {
        result = lcm (arr[0], arr[1]);
        
        for (let i = 2; i < arr.length; i++) {
            result = lcm(result, arr[i]);
        }
        
    }
    return result;
}

File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.py
def gcd(a, b):
    while b > 0:
        a, b = b, a % b
    return a

def lcd(a, b):
    return a * b // gcd(a, b)

def solution(arr):
    if len(arr) == 1: return arr[0]
    if len(arr) == 2: return lcd(arr[0], arr[1])
    result = lcd(arr[0], arr[1])
    
    for i in range(2, len(arr)):
        result = lcd(arr[i], result)
    
    return result